donate/payments/tests/test_views.py

```diff
@@ -967,3 +969,33 @@ def test_get_initial_populates_email_from_session(self):
         self.assertEqual(view.get_initial(), {
             'email': 'test@example.com',
         })
+
+
+class TestApplePayDomainAssociationView(TestCase):
+    def setUp(self):
+        self.view_url = '/.well-known/apple-developer-merchantid-domain-association'
+        self.factory = RequestFactory()
+
+    def test_view_returns_key(self):
+        """
+        Make sure the view returns the apple pay key when set
+        """
+        settings.APPLE_PAY_DOMAIN_ASSOCIATION_KEY = "test_apple_pay_key"
```

Comment:
It is recommended to not set settings directly to avoid surprising effects from test order: https://docs.djangoproject.com/en/4.1/topics/testing/tools/#overriding-settings